packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -355,6 +363,12 @@ class Auth {
         return await this.pendingToken
     }
 
+    logWarningOnClient = (silenceWarnings: boolean, msg: string) => {
+        if (onClient() && !silenceWarnings) {
+            console.warn(msg)
+        }
+    }
+
```

Comment:
Why is this called `logWarningOnClient` and not simply `logWarning`. 

Additionally, why are we passing `silenceWarnings` to a function whos sole purpose is to log a warning. I think you might be doing this to make testing easier, but I think this might be an exception to the "dependency injection is better" rule where you'll want to use `this.silenceWarnings` instead.

Reply:
@bendvc I thought of that before,  but I went with `logWarningOnClient`  because the func only logs warning on client side. You can log warning in server-side, but this funs does not do that. 🤔

Reply:
So what you have here is something like this:

```
const doSomething = (doNothing: boolean = false) => {
    if (!doNothing) {
        // do hard work here
    }
}

const main = () => {
     doSomething(true) // Whats the point of this line of code.
}
```

In general having a function where the entire content is wrapped in a conditional is a code smell, especially if that conditional is controlled by a argument that is passed to the function. If you didn't want to run the function sole logic you should just not call it.